### ISSUE TITLE
Magento 2 - laminas/laminas-server version 2.17.0 no compatible with latest module version

### ISSUE THREAD

**Issue by Ginchars** [klevu/productsearch#97]
Latest module version is not compatible with laminas/laminas-server:2.17.0
This new laminas-server version removes laminas/zend mapping and doing so breaks module as it still uses Zend\Http

For now without actually patching module is to lock laminas/laminas-server to version 2.16.0

https://github.com/klevu/productsearch/blob/8dab5ea8eb5e501e52920f94dcd7d8dee3eda83b/Model/Api/Request.php#L61

Sample use of Zend\Http classes

**Comment by sandipklevu** [comment]
Hi Thanks for raising this.

Could you share the below details for further investigation?

1. Magento version including minor version
2. PHP version
3. output of the `composer why laminas/laminas-server`
4. output of the `composer why zendframework/zend-server`

Thanks

**Comment by Ginchars** [comment]
**Magento version:** 2.4.6-p3
**PHP:** 8.1.27

**Why laminas/laminas-server when on 2.17.0:**
laminas/laminas-soap        2.13.0   requires laminas/laminas-server (^2.15) 
magento/magento2-ee-base           2.4.6-p3 requires laminas/laminas-server (^2.11) 
magento/product-enterprise-edition 2.4.6-p3 requires laminas/laminas-server (^2.11)

**Why zend-server**
composer why zendframework/zend-server

In BaseDependencyCommand.php line 78:
                                                                      
  Could not find package "zendframework/zend-server" in your project  
                                                                      
depends [-r|--recursive] [-t|--tree] [--] <package>

**Why laminas/laminas-server when locked on 2.16.0**
laminas/laminas-server             2.16.0     replaces zendframework/zend-server (^2.8.1) 
laminas/laminas-soap               2.13.0     requires laminas/laminas-server (^2.15)     
magento/magento2-ee-base           2.4.6-p3   requires laminas/laminas-server (^2.11)     
magento/product-enterprise-edition 2.4.6-p3   requires laminas/laminas-server (^2.11) 

**Why zend-server**
laminas/laminas-server 2.16.0 replaces zendframework/zend-server (^2.8.1)

**Comment by sandipklevu** [comment]
We have raised concern with the development team further.
We'll continue our communication through Freshdesk [ticket](https://klevusearch.freshdesk.com/a/tickets/82259) which was initiated.

**Comment by klevu** [comment]
Hi @Ginchars 

The 3.x branch of this module supports Magento 2.3 and 2.4. We may still have some merchants running Magento versions prior to 2.3.5 when Laminas was introduced. With this in mind, we have created a new module which any merchants running older versions can remove via Composer replace. This new module replaces Zend with Laminas. The new module is automatically included when upgrading to 3.5.0.

This new module can be found here https://github.com/klevu/api-request.
and is required here by this module here https://github.com/klevu/productsearch/blob/3.x/composer.json#L14

We have added a Knowledge Base article for this issue with an explanation as to which versions of our module will support this change going forward. https://help.klevu.com/support/solutions/articles/5000887379-http-client-change-zend-to-laminas

Kind Regards,
Richard.
